org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/StandardVMDebugger.java

```diff
@@ -314,6 +314,7 @@ public void run(VMRunnerConfiguration config, ILaunch launch, IProgressMonitor m
 			return;
 		}
 		String[] cmdLine = cmdDetails.getCommandLine();
+		System.out.println(Arrays.toString(cmdLine));
```

Comment:
production code should not log to System.out

Reply:
Sure. Did you see the commit comment?

Reply:
More to the point of this hapless attempt: do you have an idea how a debug test can log the output of the launched test program?

Two failing tests just say the process returned 1 instead of 0, but I haven't yet found a way to check if the process had produced any error message.

Reply:
> Sure. Did you see the commit comment?

no sorry,  i only read the PR message.
I don't know why the tests fail, locally it works for me.
Btw you better use `String.join(" ", cmdLine)`
For VirtualThreadsDebugViewTests build logs UnsupportedClassVersionError
![image](https://github.com/user-attachments/assets/ef2b9f2a-0e2c-41ba-8672-0c5592dfdd0d)

Reply:
> For VirtualThreadsDebugViewTests build logs UnsupportedClassVersionError

That's interesting, but also very strange.

The test uses the `IJavaProject` created by `org.eclipse.jdt.debug.tests.AbstractDebugTest.get24Project()`, which clearly sets compliance to 24. So who created a classfile with version 67 aka "23"??

Reply:
See also https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3605

Reply:
On top of all that, why is VirtualThreadsDebugViewTests enabling preview? True, I was the last one changing it (to 24), but before that it used 23 plus preview ...

Btw, I'd love to read some background in the PR that introduced that class, but I can't find that PR...

Reply:
956d6682e76431709e4cda00c22599754d62f958
![image](https://github.com/user-attachments/assets/844b29c7-15c0-4b61-bd16-afc8a34e30b3)
-> https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/535

Reply:
I only wish such a link were available from EGit ... but I see that lack of a link in the commit message was already complained about ...